helpers/env-variables.ts

```diff
@@ -423,9 +424,29 @@ const getToolEnvs = (tools?: Tool[]): EnvVar[] => {
   return toolEnvs;
 };
 
-const getSystemPromptEnv = (tools?: Tool[]): EnvVar => {
-  const defaultSystemPrompt =
-    "You are a helpful assistant who helps users with their questions.";
+const getSystemPromptEnv = (
+  tools?: Tool[],
+  dataSources?: TemplateDataSource[],
+): EnvVar => {
+  let defaultSystemPrompt = "";
+  if (dataSources?.find((ds) => ds.type === "file" || ds.type === "web")) {
+    defaultSystemPrompt = `'You\'re a helpful assistant! You have provided information from a knowledge base that has been passed to you in nodes of information.
```

Comment:
1. why not appending this to the `defaultSystemPrompt` if we have data sources?
```suggestion
    defaultSystemPrompt = `You have provided information from a knowledge base that has been passed to you in nodes of information.
```
2. does the prompt work with queryenginetool?

Reply:
1. Updated.
2. I didn't check but it should work because this is a synthesis prompt which applied after retrieval step.